source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,266 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding a subset of them in memory at a given
+time. The Go driver uses a cursor in some read operations to return
+matched documents in batches as opposed to all at once.
+
+Retrieve Sample Data
+~~~~~~~~~~~~~~~~~~~~
+
+Each section assumes you retrieve all the documents in your collection
+with the ``Find()`` function.
```

Comment:
Suggestion: As the examples on this page do not reference a sample data set, could "Retrieve Sample Data" be changed to reflect the "sample cursor" examples reference?

Rough draft of potential solution:

```suggestion
Example Cursor
~~~~~~~~~~~~

The sections in this guide reference the `cursor` variable from the following code snippet. The cursor variable contains a cursor that contains all documents in a collection.
```